airbyte-db/lib/src/main/java/io/airbyte/db/instance/FlywayMigrationDatabase.java

```diff
@@ -32,9 +37,15 @@ public abstract class FlywayMigrationDatabase extends PostgresDatabase {
 
   private Connection connection;
 
-  protected abstract Database getAndInitializeDatabase(String username, String password, String connectionString) throws IOException;
+  protected abstract Database getAndInitializeDatabase(DSLContext dslContext) throws IOException;
 
-  protected abstract DatabaseMigrator getDatabaseMigrator(Database database);
+  protected abstract DatabaseMigrator getDatabaseMigrator(Database database, Flyway flyway);
+
+  protected abstract String getInstalledBy();
```

Comment:
are these changes new additions>

Reply:
@davinchia Yes.  Originally, the `Flyway` object was created internal to the `FlywayDatabaseMigrator` class.  By changing it so that the constructed `Flyway` object is passed to the migrator (and its sub classes), these `*MigrationDatabase` classes also needed to change to be able to get access to the data needed to construct the `Flyway` object.  Put another way, the inversion of control means that we now need to be able to get the parameters  needed to construct the two different `Flyway` objects (one for configs and one for jobs).  These new methods push the implementation down to the configs/jobs implementations of the migration database to provide the correct data.

Reply:
I understand. thanks for explaining.